components/buildless-serverless/internal/controller/git/commit_checker.go

```diff
@@ -4,23 +4,33 @@ import (
 	"github.com/go-git/go-git/v5"
 	"github.com/go-git/go-git/v5/plumbing"
 	"github.com/go-git/go-git/v5/storage/memory"
+	"github.com/pkg/errors"
 )
 
 //go:generate mockery --name=LastCommitChecker --output=automock --outpkg=automock --case=underscore
 type LastCommitChecker interface {
-	GetLatestCommit(url, reference string) (string, error)
+	GetLatestCommit(url, reference string, gitAuth *GitAuth) (string, error)
 }
 
 type GoGitCommitChecker struct {
 }
 
-func (g GoGitCommitChecker) GetLatestCommit(url, reference string) (string, error) {
-	r, err := git.Clone(memory.NewStorage(), nil, &git.CloneOptions{
+func (g GoGitCommitChecker) GetLatestCommit(url, reference string, gitAuth *GitAuth) (string, error) {
```

Comment:
If I understand correctly, this code will always clone a repo, even if the reference value is HASH. Can we somehow detect if the reference is set to HASH and just return it? to not clone the whole repo to return data we got from the argument

Reply:
The Reference is only the string and I can't see how to check if it's hash or branch. 
I checked the old solution and there we do the same - clone repo and then get the commit.